src/controllers/orderController.js

```
@@ -0,0 +1,135 @@
+const { NotFoundError, BadRequestError } = require('../errors');
+const { StatusCodes } = require('http-status-codes');
+const { calculateCartTotal } = require('../utils/cartTotal');
+
+const getOrders = async (req, res, next) => {
+    try {
+        const userId = req.user.userId;
+        const buyOrders = await Order.find({ buyer: userId });
+        const sellOrders = await Order.find({ seller: userId }).populate('buyer', 'firstName lastName email');
+        res.status(StatusCodes.OK).json({ buyOrders, sellOrders });
+    } catch (error) {
+        next(error);
+    }
+};
+
+const createOrderFromCart = async (req, res, next) => {
+    try {
+        const userId = req.user.userId;
+        const cart = await Cart.findOne({ createdBy: userId }).populate('orderItems.book');
+
+        if (!cart || cart.orderItems.length === 0) {
+            throw new BadRequestError('Your cart is empty');
+        }
+
+        const itemsBySeller = cart.orderItems.reduce((acc, item) => {
+            const sellerId = item.book.createdBy.toString();
+            if (!acc[sellerId]) acc[sellerId] = [];
+            acc[sellerId].push(item);
+            return acc;
+        }, {});
+
+        const orders = [];
+        for (const [sellerId, items] of Object.entries(itemsBySeller)) {
+            const orderItems = items.map((item) => ({
+                book: {
+                    title: item.book.title,
+                    author: item.book.author,
+                    condition: item.book.condition,
+                },
+                price: item.price,
+            }));
+
+            const tempCart = { orderItems };
+            const buyerLocation = req.user.location;
+            const sellerLocation = items[0].book.createdBy.location;
+            const orderAmount = calculateCartTotal(tempCart, buyerLocation, sellerLocation);
+
+            const order = await Order.create({
+                orderNumber: `ORD-${Date.now()}-${sellerId.slice(-5)}`,
```

can you test that invalid data can't throw an error here, like a missing createdBy(if we are validating that, then it should never be missing so its pretty solid, but always good to check.